src/frontends/tensorflow_common/src/op/strided_slice.cpp

```diff
@@ -16,42 +16,55 @@ namespace tensorflow {
 namespace op {
 
 OutputVector translate_strided_slice_op(const NodeContext& node) {
+    default_op_checks(node, 4, {"StridedSlice", "STRIDED_SLICE"});
     auto input = node.get_input(0);
     auto begin = node.get_input(1);
     auto end = node.get_input(2);
     auto strides = node.get_input(3);
 
-    auto begin_mask = node.get_attribute<int64_t>("begin_mask", 0);
-    auto end_mask = node.get_attribute<int64_t>("end_mask", 0);
-    auto new_axis_mask = node.get_attribute<int64_t>("new_axis_mask", 0);
-    auto ellipsis_mask = node.get_attribute<int64_t>("ellipsis_mask", 0);
-    auto shrink_axis_mask = node.get_attribute<int64_t>("shrink_axis_mask", 0);
-
     auto mask_to_vector = [](int64_t mask) {
-        size_t length = sizeof(mask) * CHAR_BIT;
-        vector<int64_t> vec(length, 0);
         if (mask == 0) {
-            return vec;
+            return vector<int64_t>{};
         }
-        for (size_t i = 0; i < length; ++i) {
-            if (static_cast<unsigned char>(mask >> i & 0x1) == 1) {
+        size_t max_length = sizeof(mask) * CHAR_BIT;
+        vector<int64_t> vec{};
+        for (size_t i = 0; i < max_length; ++i) {
+            if ((mask >> i & 0x1) == 1) {
+                // resize the vector by appending with required number of zeros
+                vec.resize(i + 1, 0);
                 vec[i] = 1;
             }
         }
         return vec;
     };
 
-    auto res = make_shared<StridedSlice>(input,
-                                         begin,
-                                         end,
-                                         strides,
-                                         mask_to_vector(begin_mask),
-                                         mask_to_vector(end_mask),
-                                         mask_to_vector(new_axis_mask),
-                                         mask_to_vector(shrink_axis_mask),
-                                         mask_to_vector(ellipsis_mask));
-    set_node_name(node.get_name(), res);
-    return res->outputs();
+    // retrieve attributes for StridedSlice operation
+    auto begin_mask = mask_to_vector(node.get_attribute<int64_t>("begin_mask", 0));
+    auto end_mask = mask_to_vector(node.get_attribute<int64_t>("end_mask", 0));
+    auto new_axis_mask = mask_to_vector(node.get_attribute<int64_t>("new_axis_mask", 0));
+    auto ellipsis_mask = mask_to_vector(node.get_attribute<int64_t>("ellipsis_mask", 0));
+    auto shrink_axis_mask = mask_to_vector(node.get_attribute<int64_t>("shrink_axis_mask", 0));
+
+    // the masks can be of different length and we need to align them by the maximum length
+    size_t max_length = std::max(
+        {begin_mask.size(), end_mask.size(), new_axis_mask.size(), ellipsis_mask.size(), shrink_axis_mask.size()});
+    begin_mask.resize(max_length, 0);
```

Comment:
Strange place, why don't reserve same size for each mask? They looks like may have same maximum size (sizeof(int64) * CHAR_BIT). And if we want to use vector.size() as something meaningful - resize to less is preferred than possible increase.

Reply:
If we reserve maximum size `size (sizeof(int64) * CHAR_BIT)`, IRs look ugly due to a lot of zeros as currently they do and we would like to avoid it.

Reply:
What's the difference in IR's between "allocate minimal" -> "resize to max_length (possible reallocation)" and "allocate sizeof(int64) * CHAR_BIT" -> "resize to max_length (no reallocation)"?

Reply:
Here is the difference:
![image](https://user-images.githubusercontent.com/35459624/226297699-2425d7ac-68fa-4e5d-984f-888e7e469773.png)